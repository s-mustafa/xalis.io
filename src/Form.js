import React, { useState } from 'react'

const INITIAL_STATE = {
  desc: '',
  amount: '',
  date: '',
}

const Form = ({ onAdd }) => {
  const [formData, setFormData] = useState({ ...INITIAL_STATE })
  const onChange = name => e => {
    setFormData({ ...formData, [name]: e.target.value })
  }

  const onSubmit = e => {
    e.preventDefault()

    onAdd(formData)
    setFormData({ ...INITIAL_STATE })
  }
  return (
    <form className="py-1 my-4 mx-auto w-2/3" onSubmit={onSubmit}>
      <p className="border-b mb-3 py-2 font-bold">Ajouter une transaction</p>
      <div className="grid grid-cols-1 sm:grid-cols-3 gap-4">
        <input
          name="desc"
          value={formData.desc}
          onChange={onChange('desc')}
          className="bg-white focus:outline-none focus:shadow-outline border border-gray-300 rounded-lg py-2 px-4 block w-full appearance-none leading-normal"
          type="text"
          placeholder="Description"
          data-testid="description"
          autoComplete="off"
        />
        <input
          name="amount"
          value={formData.amount}
          onChange={onChange('amount')}
          className="bg-white focus:outline-none focus:shadow-outline border border-gray-300 rounded-lg py-2 px-4 block w-full appearance-none leading-normal"
          type="number"
          data-testid="amount"
          placeholder="Montant"
        />
        <input
          name="date"
          value={formData.date}
          onChange={onChange('date')}
          className="bg-white focus:outline-none focus:shadow-outline border border-gray-300 rounded-lg py-2 px-4 block w-full appearance-none leading-normal"
          type="date"
          data-testid="date"
          placeholder="Date"
        />
      </div>
      <button
        type="submit"
        data-testid="submit-button"
        className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 mt-3 rounded-lg">
        Ajouter
      </button>
    </form>
  )
}

export default Form
