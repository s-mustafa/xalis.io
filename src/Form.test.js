import React from 'react'
import Form from './Form'
import renderer from 'react-test-renderer'

test('Form renders correctly', () => {
  const component = renderer.create(<Form />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
