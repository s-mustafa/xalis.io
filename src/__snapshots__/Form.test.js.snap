// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders correctly 1`] = `
<form
  className="py-1 my-4 mx-auto w-2/3"
  onSubmit={[Function]}
>
  <p
    className="border-b mb-3 py-2 font-bold"
  >
    Ajouter une transaction
  </p>
  <div
    className="grid grid-cols-1 sm:grid-cols-3 gap-4"
  >
    <input
      autoComplete="off"
      className="bg-white focus:outline-none focus:shadow-outline border border-gray-300 rounded-lg py-2 px-4 block w-full appearance-none leading-normal"
      data-testid="description"
      name="desc"
      onChange={[Function]}
      placeholder="Description"
      type="text"
      value=""
    />
    <input
      className="bg-white focus:outline-none focus:shadow-outline border border-gray-300 rounded-lg py-2 px-4 block w-full appearance-none leading-normal"
      data-testid="amount"
      name="amount"
      onChange={[Function]}
      placeholder="Montant"
      type="number"
      value=""
    />
    <input
      className="bg-white focus:outline-none focus:shadow-outline border border-gray-300 rounded-lg py-2 px-4 block w-full appearance-none leading-normal"
      data-testid="date"
      name="date"
      onChange={[Function]}
      placeholder="Date"
      type="date"
      value=""
    />
  </div>
  <button
    className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 mt-3 rounded-lg"
    data-testid="submit-button"
    type="submit"
  >
    Ajouter
  </button>
</form>
`;
